src/p4est_io.h

```diff
@@ -68,6 +108,398 @@ p4est_t            *p4est_inflate (sc_MPI_Comm mpicomm,
+ *                              It is possible, however, to use a different
+ *                              partition or number of ranks from writing it.
+ * \param [in] filename         The path to the file that is opened.
+ * \param [in,out] user_string  At least 16 bytes. The user string is written
+ *                              to the passed array including padding spaces
+ *                              and a trailing null-termination.
+ * \param [out] errcode         An errcode that can be interpreted by \ref
+ *                              p4est_file_error_string and
+ *                              \ref p4est_file_error_class.
+ * \return                      Newly allocated context to continue reading
+ *                              and eventually closing the file. NULL in
+ *                              case of error.
+ */
+p4est_file_context_t *p4est_file_open_read (p4est_t * p4est,
+                                            const char *filename,
+                                            char *user_string, int *errcode);
+
+/** Write a header block to an opened file.
+ * This function requires an opened file context.
+ * The header data and its metadata are written on rank 0.
+ *
+ * \param [out] fc            Context previously created by \ref
+ *                            p4est_file_open_create.
+ * \param [in]  header_size   The size of header_data in bytes.
+ *                            This function returns the passed fc
+ *                            parameter and sets errcode to
+ *                            sc_MPI_SUCCESS if it is called
+ *                            for header_size == 0. Nothing is
+ *                            written to the file and fc stays
+ *                            untouched.
+ * \param [in]  header_data   A pointer to the header data. The user is
+ *                            responsible for the validality of the header
+ *                            data.
+ * \param [in]  user_string   Maximal 47 bytes. These chars are written to
+ *                            the block header and padded to 47 chars by adding
+ *                            spaces.
+ * \param [out] errcode       An errcode that can be interpreted by \ref
+ *                            p4est_file_error_string and
+ *                            \ref p4est_file_error_class.
+ * \return                    Return the input context to continue writing
+ *                            and eventually closing the file. The return
+ *                            value is NULL in case of error, then
+ *                            it also holds errcode != 0 and the file is
+ *                            tried to close and fc is freed.
+ */
+p4est_file_context_t *p4est_file_write_header (p4est_file_context_t * fc,
+                                               size_t header_size,
+                                               const void *header_data,
+                                               const char user_string[47],
```

Comment:
As above, please add space for the terminating `NUL` (so 48 characters) and decide on using a `#define` consistently.

Reply:
> As above, please add space for the terminating `NUL` (so 48 characters) and decide on using a `#define` consistently.

This is done as described above.

Reply:
> As above, please add space for the terminating `NUL` (so 48 characters) and decide on using a `#define` consistently.

The null-termination of string is done as described above. The defines for the number of bytes of a user string is now used consistently.